vpr/src/route/bucket.h

```diff
@@ -0,0 +1,226 @@
+#ifndef _BUCKET_H
+#define _BUCKET_H
+
+#include <vector>
+
+#include "heap_type.h"
+
```

Comment:
Please add a high-level comment on how the bucket sort works, particularly the two levels of allocation (large, reusable memory blocks are pre-allocated and managed, and individually allocated and released out to callers heap items).  Define terms for the two types of references/storage being tracked (suggest pre-allocated and allocated, or maybe group-allocated and fully-allocated).

Reply:
Done